### PR TITLE
Avoid recursive runInTransaction calls with relations

### PR DESCRIPTION
**Pull request by vaind**

### CODE REVIEW

File: objectbox/lib/src/transaction.dart [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  }
+
+  /// Executes a given function inside a transaction.
+  ///
+  /// Returns type of [fn] if [return] is called in [fn].
+  static R execute<R>(Store store, TxMode mode, R Function() fn) {
+    final tx = Transaction(store, mode);
+    try {
+      if (tx._isWrite) tx.markSuccessful(true);
+      return fn();
```
Comment (greenrobot-team) [inline objectbox/lib/src/transaction.dart:76]: I know this is the same behavior as before, but shouldn't the transaction be marked successful *after* `fn` executed? E.g. similar to `put()`?

File: objectbox/lib/src/relations/to_many.dart [hunk 2]
```diff
@@ -201,7 +204,10 @@ class ToMany<EntityT> extends Object with ListMixin<EntityT> {
             throw UnimplementedError();
         }
       });
-    });
+      tx.markSuccessful(true);
```
Comment (greenrobot-team) [inline objectbox/lib/src/relations/to_many.dart:207]: Also `if (ownedTx)`? Again, I'm not sure how markSuccessful works with ObjectBox.

**Review by greenrobot-team — COMMENTED** [review]
Some questions, otherwise LGTM.

File: objectbox/lib/src/transaction.dart (hunk 1 above)
Comment (vaind) [reply objectbox/lib/src/transaction.dart:76]: Yes, in the perfect world where we wouldn't have to define a variable storing the result of `fn()` before returning it after a `markSuccessful(true)` call. This is more efficient, assuming transactions "usually" don't fail

File: objectbox/lib/src/relations/to_many.dart (hunk 2 above)
Comment (vaind) [reply objectbox/lib/src/relations/to_many.dart:207]: you can call it multiple times, but yes, no need to do it here if not owned.

File: objectbox/lib/src/transaction.dart (hunk 1 above)
Comment (greenrobot-team) [reply objectbox/lib/src/transaction.dart:76]: So this is intended, maybe add a note?

Comment (vaind) [reply objectbox/lib/src/transaction.dart:94]: OK

**Pull request merged**